SlipeServer.Packets/Definitions/Ped/PedSyncPacket.cs

```diff
@@ -0,0 +1,158 @@
+        public override PacketPriority Priority { get; } = PacketPriority.Medium;
+
+        public struct SyncData
+        {
+            public bool Send { get; set; }
+            public uint Model { get; set; }
+            public byte Flags { get; set; }
+            public byte TimeSyncContext { get; set; }
+            public Vector3 Position { get; set; }
+            public float Rotation { get; set; }
+            public Vector3 Velocity { get; set; }
+            public float Health { get; set; }
+            public float Armor { get; set; }
+            public bool IsOnFire { get; set; }
+            public bool IsInWater { get; set; }
+        }
+
+        private List<SyncData> Syncs;
+
+        public PedSyncPacket()
+        {
+            this.Syncs = new List<SyncData>();
+        }
+
+        public override byte[] Write()
+        {
+            var builder = new PacketBuilder();
+
+            foreach (var data in this.Syncs)
+            {
+                if (data.Send)
+                {
+                    // Vehicle ID
+                    builder.Write(data.Model);
+
+                    // Sync time context
+                    builder.Write(data.TimeSyncContext);
+
+                    // Flags
+                    builder.Write(data.Flags);
+
+                    // Position
+                    if ((data.Flags & 0x01) != 0)
+                    {
+                        builder.Write(data.Position.X);
+                        builder.Write(data.Position.Y);
+                        builder.Write(data.Position.Z);
+                    }
+
+                    // Rotation
+                    if ((data.Flags & 0x02) != 0)
+                        builder.Write(data.Rotation);
+                    
+
+                    // Velocity
+                    if ((data.Flags & 0x04) != 0)
+                    {
+                        builder.Write(data.Velocity.X);
+                        builder.Write(data.Velocity.Y);
+                        builder.Write(data.Velocity.Z);
+                    }
+
+                    // Health, armor, on fire, is in water
+                    if ((data.Flags & 0x08) != 0)
+                        builder.Write(data.Health);
+                    if ((data.Flags & 0x10) != 0)
+                        builder.Write(data.Armor);
+                    if ((data.Flags & 0x20) != 0)
+                        builder.Write(data.IsOnFire);
+                    if ((data.Flags & 0x40) != 0)
+                        builder.Write(data.IsInWater);
+                }
+            }
+
+            return builder.Build();
+        }
+
+        public override void Read(byte[] bytes)
+        {
+            var reader = new PacketReader(bytes);
+            while ((reader.Size / 8) > 32)
+            {
+                SyncData data = new SyncData();
+                data.Send = false;
+
+                data.Model = reader.GetByte();
```

Comment:
I don't think the Element ID is contained in just 1 byte?
Afaik it's uint so 2 bytes (you also defined it as uint in SyncData :))

```suggestion
                data.Model = reader.GetUint16();
```